### PR TITLE
OnBoarding: Authentication Flow

### PR DESCRIPTION
**Pull request by Rajan-p-simform**
- Login Screen
- Register Screen
- Forget Password Screen

### CODE REVIEW

File: app/src/main/java/simform/gitexcercise/authentication/ForgetPasswordFragment.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    lateinit var binding: FragmentForgetPasswordBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = DataBindingUtil.inflate(layoutInflater,R.layout.fragment_forget_password,container,false)
+        return binding.root
+    }
+
```
Comment (nishchal-v) [inline app/src/main/java/simform/gitexcercise/authentication/ForgetPasswordFragment.kt]: Remove the empty line here

File: app/src/main/java/simform/gitexcercise/authentication/LoginFragment.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        binding.buttonGoToRegister.setOnClickListener {
+            Navigation.findNavController(view).navigate(R.id.action_loginFragment_to_registrationFragment)
+        }
+        binding.textForgetPassword.setOnClickListener {
+            Navigation.findNavController(view).navigate(R.id.action_loginFragment_to_forgetPasswordFragment)
+        }
+
+    }
+
```
Comment (nishchal-v) [inline app/src/main/java/simform/gitexcercise/authentication/LoginFragment.kt]: Same here as above comment

File: app/src/main/java/simform/gitexcercise/authentication/RegistrationFragment.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.navigation.Navigation
+import simform.gitexcercise.android.R
+import simform.gitexcercise.android.databinding.FragmentRegistrationBinding
+
+class RegistrationFragment : Fragment() {
+
+    private lateinit var binding: FragmentRegistrationBinding
+    override fun onCreateView(
```
Comment (nishchal-v) [inline app/src/main/java/simform/gitexcercise/authentication/RegistrationFragment.kt:23]: Add a empty line above

File: app/src/main/java/simform/gitexcercise/authentication/LoginFragment.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    ): View {
+        binding = DataBindingUtil.inflate(layoutInflater,R.layout.fragment_login,container,false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.buttonGoToRegister.setOnClickListener {
+            Navigation.findNavController(view).navigate(R.id.action_loginFragment_to_registrationFragment)
```
Comment (nishchal-v) [inline app/src/main/java/simform/gitexcercise/authentication/LoginFragment.kt]: We can have a lazy variable of this navigation contorller

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,3 +1,15 @@
 <resources>
     <string name="app_name">GitExcercise</string>
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (nishchal-v) [inline app/src/main/res/values/strings.xml]: Remove this demo code

**Review by nishchal-v — CHANGES_REQUESTED** [review]
- Formate the files properly with lines where required and remove extra lines
- Change the title and the commit message as per our standards

**Comment by Rajan-p-simform** [comment]
> * Formate the files properly with lines where required and remove extra lines
> * Change the title and the commit message as per our standards

**Review by nishchal-v — APPROVED** [review]

**Pull request merged**